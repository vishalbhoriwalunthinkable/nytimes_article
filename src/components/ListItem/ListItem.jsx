import React from "react";
import PropTypes from "prop-types";
import useMediaQuery from "@material-ui/core/useMediaQuery";

import {
  Avatar,
  IconButton,
  ListItemAvatar,
  ListItemSecondaryAction,
  ListItemText,
  makeStyles,
  ListItem,
} from "@material-ui/core";
import { ChevronRight, Today } from "@material-ui/icons";

const useStyles = makeStyles((theme) => ({
  container: {
    width: "100%",
  },
  root: {
    width: "100%",
    maxWidth: "850px",
    maxHeight: "40px",
    margin: "0 auto",
    alignItems: "center",
    wordBreak: "break-text",
    justifyContent: "center",
    backgroundColor: theme.palette.background.paper,
  },
  listItem: {
    display: "flex",
    minWidth: "100%",
    justifyContent: "space-between",
  },
  date: {
    marginTop: "4px",
  },
  link: {
    textDecoration: "none",
    color: "#000",
  },
  listitemtext: {
    wordBreak: "break-word",
  },
}));

const ListItemComponent = ({ data }) => {
  const matches = useMediaQuery("(min-width:600px)");
  const classes = useStyles();
  const { abstract, byline, url, published_date } = data || {};

  return (
    <a href={url} className={classes.link} target="_blank" rel="noreferrer">
      <ListItem className={classes.listItem} alignItems="flex-start">
        <ListItemAvatar>
          <Avatar />
        </ListItemAvatar>
        <ListItemText
          style={{ width: !matches ? "210px" : null, textAlign: "justify" }}
          primary={<div className="ellipsis">{abstract || "NA"}</div>}
          secondary={
            <div
              style={{
                display: matches ? "flex" : "inline",
                justifyContent: "space-between",
              }}
            >
              <div>{byline}</div>
              <div variant="body2" className={classes.date} color="textPrimary">
                <div
                  style={{
                    display: "flex",
                    width: "fit-content",
                  }}
                >
                  <Today fontSize="small" />
                  <div>{published_date || "NA"}</div>
                </div>
              </div>
            </div>
          }
          className={classes.listitemtext}
        />
        <ListItemSecondaryAction style={{ right: "0" }}>
          <IconButton edge="end" aria-label="comments">
            <ChevronRight fontSize="large" />
          </IconButton>
        </ListItemSecondaryAction>
      </ListItem>
    </a>
  );
};

ListItemComponent.defaultProps = {
  data: {},
};

ListItemComponent.propTypes = {
  data: PropTypes.object,
};

export default ListItemComponent;
