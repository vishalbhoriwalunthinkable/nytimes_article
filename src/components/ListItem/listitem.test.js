import React from "react";
import renderer from "react-test-renderer";

import ListItem from "./ListItem";

it("ListItem renders without props", () => {
  const tree = renderer.create(<ListItem />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("ListItem renders with props", () => {
  const data = {
    abstract: "abstract",
    byline: "byline",
    published_date: "data",
  };
  const tree = renderer.create(<ListItem data={data} />).toJSON();
  expect(tree).toMatchSnapshot();
});
