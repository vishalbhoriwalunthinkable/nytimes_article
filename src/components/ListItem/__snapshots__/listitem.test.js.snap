// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders with props 1`] = `
<a
  className="makeStyles-link-5"
  rel="noreferrer"
  target="_blank"
>
  <li
    className="MuiListItem-container"
  >
    <div
      className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters MuiListItem-alignItemsFlexStart MuiListItem-secondaryAction"
      disabled={false}
    >
      <div
        className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiAvatar-fallback"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiListItemText-root makeStyles-listitemtext-6 MuiListItemText-multiline"
        style={
          Object {
            "textAlign": "justify",
            "width": "210px",
          }
        }
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          <div
            className="ellipsis"
          >
            abstract
          </div>
        </span>
        <p
          className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          <div
            style={
              Object {
                "display": "inline",
                "justifyContent": "space-between",
              }
            }
          >
            <div>
              byline
            </div>
            <div
              className="makeStyles-date-4"
              color="textPrimary"
              variant="body2"
            >
              <div
                style={
                  Object {
                    "display": "flex",
                    "width": "fit-content",
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zM7 10h5v5H7z"
                  />
                </svg>
                <div>
                  data
                </div>
              </div>
            </div>
          </div>
        </p>
      </div>
    </div>
    <div
      className="MuiListItemSecondaryAction-root"
      style={
        Object {
          "right": "0",
        }
      }
    >
      <button
        aria-label="comments"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </li>
</a>
`;

exports[`ListItem renders without props 1`] = `
<a
  className="makeStyles-link-5"
  rel="noreferrer"
  target="_blank"
>
  <li
    className="MuiListItem-container"
  >
    <div
      className="MuiListItem-root makeStyles-listItem-3 MuiListItem-gutters MuiListItem-alignItemsFlexStart MuiListItem-secondaryAction"
      disabled={false}
    >
      <div
        className="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiAvatar-fallback"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiListItemText-root makeStyles-listitemtext-6 MuiListItemText-multiline"
        style={
          Object {
            "textAlign": "justify",
            "width": "210px",
          }
        }
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          <div
            className="ellipsis"
          >
            NA
          </div>
        </span>
        <p
          className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          <div
            style={
              Object {
                "display": "inline",
                "justifyContent": "space-between",
              }
            }
          >
            <div />
            <div
              className="makeStyles-date-4"
              color="textPrimary"
              variant="body2"
            >
              <div
                style={
                  Object {
                    "display": "flex",
                    "width": "fit-content",
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zM7 10h5v5H7z"
                  />
                </svg>
                <div>
                  NA
                </div>
              </div>
            </div>
          </div>
        </p>
      </div>
    </div>
    <div
      className="MuiListItemSecondaryAction-root"
      style={
        Object {
          "right": "0",
        }
      }
    >
      <button
        aria-label="comments"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </span>
      </button>
    </div>
  </li>
</a>
`;
